class.jetpack.php

```diff
@@ -3872,6 +3876,31 @@ function disconnect_survey_notice() {
 		<?php
 	}
 
+	/**
+	 * Checks if the user have started a register flow from calypso in the last day, and if so, redirect them there
+	 */
+	function check_jetpack_connect_redirection() {
+		$user = $this->get_connected_user_data();
+		$parsed_site = parse_url( get_site_url() );
+
+		if ( isset( $user[ 'jetpack_connect' ] ) && is_array( $user[ 'jetpack_connect' ] ) ) {
+			foreach ($user[ 'jetpack_connect' ] as $jetpack_connect_request ) {
+				$parsed_jetpack_connect_site = $jetpack_connect_request[ 'site_url' ];
+
+				if ( $parsed_jetpack_connect_site->domain == $parsed_site->domain
+					&& $parsed_jetpack_connect_site->path == $parsed_site->path
+					&& ( time() - $this->JETPACK_CONNECT_TIMEOUT ) < strtotime( $jetpack_connect_request[ 'date' ] ) ) {
+					// the user started a flow from calypso registering this site url in the last 24 hours
+					$this->message .= __( '<div> Let us take you back to WordPress.com.</div>' );
+					echo '<script> setTimeout( function() { window.location = "' . $this->JETPACK_CONNECT_FLOW_URL . $this->build_raw_urls( get_site_url() ) . '" }, 1000 ); </script>';
```

Comment:
yeah, that way the "You're fueled up and ready to go, Jetpack is now active." message wouldn't be shown, because you get redirected directly back to calypso without rendering the current page. I thought it would be nicer to show the message and the redirect after one second (not sure if there's a way to do this with any native method so we can avoid that nasty chunk of js, though)